iamdata.tf

```diff
@@ -0,0 +1,74 @@
+data "aws_iam_policy_document" "iam_group_with_user_self_service" {
+  statement {
+    sid = "ListUsersAccessWithoutMFA"
+
+    actions = [
+      "iam:ListVirtualMFADevices"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    sid = "SelfServiceAccessWithoutMFA"
+
+    actions = [
+        "iam:List*",
+        "iam:GetUser",
+        "iam:GetAccountPasswordPolicy",
+        "iam:ChangePassword",
+        "iam:CreateVirtualMFADevice",
+        "iam:EnableMFADevice"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:iam::*:user/$${aws:username}",
+      "arn:aws:iam::*:mfa/$${aws:username}"
+    ]
+  }
+
+  statement {
+    sid = "SelfServiceAccessWithMFA"
+
+    actions = [
+        "iam:Get*",
+        "iam:DeleteSSHPublicKey",
+        "iam:GetSSHPublicKey",
+        "iam:ListSSHPublicKeys",
+        "iam:UpdateSSHPublicKey",
+        "iam:UploadSSHPublicKey",
+        "iam:CreateAccessKey",
+        "iam:DeleteAccessKey",
+        "iam:UpdateAccessKey",
+        "iam:DeleteVirtualMFADevice",
+        "iam:DeactivateMFADevice",
+        "iam:ResyncMFADevice",
+        "iam:UploadSigningCertificate",
+        "iam:UpdateSigningCertificate",
+        "iam:DeleteSigningCertificate",
+        "iam:GenerateServiceLastAccessedDetails"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:iam::*:user/$${aws:username}",
+      "arn:aws:iam::*:mfa/$${aws:username}"
+    ]
+
+    condition {
+      test     = "BoolIfExists"
```

Comment:
I think this maybe should be `bool`
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html#condition-keys-multifactorauthpresent

Reply:
@rchildress87 That document says `BoolIfExists` is correct, the policy we're using is the AWS published one that everyone uses.

We recommend that you use the [BoolIfExists](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_IfExists) operator to check whether a request is authenticated using MFA.

Reply:
I think one of the `bool` example under Alternative Combinations should be considered as better option but, according to above documentation, this does emulate the policy that is created when you use the IAM visual editor and choose *MFA Required*. It also matches the policy that this data source is replacing. I'll approve.